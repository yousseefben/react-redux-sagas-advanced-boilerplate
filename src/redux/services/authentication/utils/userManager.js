import { createUserManager } from 'redux-oidc';

const userManagerConfig = {
  client_id: 'example-app',
  redirect_uri: `${window.location.protocol}//${window.location.hostname}${
    window.location.port ? `:${window.location.port}` : ''
  }/callback`,
  response_type: 'token id_token',
  // scope: 'openid profile https://www.googleapis.com/auth/youtube.readonly',
  authority: 'http://localhost:8080/auth/realms/test',
  silent_redirect_uri: `${window.location.protocol}//${
    window.location.hostname
  }${window.location.port ? `:${window.location.port}` : ''}/silent_renew.html`,
  automaticSilentRenew: true,
  filterProtocolClaims: true,
  loadUserInfo: true
};

const userManager = createUserManager(userManagerConfig);

export default userManager;
