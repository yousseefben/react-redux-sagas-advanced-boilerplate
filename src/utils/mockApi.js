export const mockSuccesfulResponse = (
  status = 200,
  method,
  returnBody,
  contentType
) => {
  global.fetch = jest.fn().mockImplementationOnce(
    () =>
      new Promise(resolve => {
        resolve({
          headers: {
            get: () => {
              return contentType || 'application/json';
            }
          },
          ok: true,
          status,
          json: () => returnBody || {},
          text: () => returnBody
        });
      })
  );
};

export const mockFailedResponse = error => {
  global.fetch = jest.fn().mockImplementationOnce(
    () =>
      new Promise(() => {
        throw new Error(error);
      })
  );
};
