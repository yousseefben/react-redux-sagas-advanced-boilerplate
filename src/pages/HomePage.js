import React from 'react';
import { Button } from 'antd';
import userManager from '../redux/services/authentication/utils/userManager';

const ExampleComponent = React.lazy(() =>
  import('../components/Example/ExampleComponent')
);

const HomePage = () => (
  <div data-testid="home">
    hello
    <React.Suspense fallback={<div>Loading ...</div>}>
      <ExampleComponent />
      <Button
        data-testid="btn-login"
        onClick={() => {
          userManager.signinRedirect();
        }}
      >
        Log In
      </Button>
    </React.Suspense>
  </div>
);
export default HomePage;
