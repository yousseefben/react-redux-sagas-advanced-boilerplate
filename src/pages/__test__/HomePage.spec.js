import React from 'react';

import { fireEvent, cleanup } from '@testing-library/react';
import HomePage from '../HomePage';
import { renderWithRedux } from '../../utils/test_utils';
import userManager from '../../redux/services/authentication/utils/userManager';

afterEach(cleanup);

test('should render HomePage', async () => {
  const component = renderWithRedux(<HomePage />);

  expect(component.baseElement).toMatchSnapshot();
});

test('should call signinRedirect', async () => {
  const { getByTestId } = renderWithRedux(<HomePage />);

  jest.spyOn(userManager, 'signinRedirect');

  fireEvent.click(getByTestId('btn-login'));
  expect(userManager.signinRedirect).toHaveBeenCalledTimes(1);
});
