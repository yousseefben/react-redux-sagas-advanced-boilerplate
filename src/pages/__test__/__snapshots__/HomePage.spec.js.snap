// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HomePage 1`] = `
<body>
  <div>
    <div
      data-testid="home"
    >
      hello
      <button
        class="ant-btn"
        data-testid="btn-login"
        style="display: none;"
        type="button"
      >
        <span>
          Log In
        </span>
      </button>
      <div>
        Loading ...
      </div>
    </div>
  </div>
</body>
`;
